src/protocols/arp.rs

```diff
@@ -15,16 +16,28 @@ pub struct ArpHeader {
   pub op_code: u16,
 }
 
+impl ArpHeader {
+  pub fn from_bytes(bytes: &[u8]) -> Option<ArpHeader> {
+    if bytes.len() == size_of::<ArpHeader>() {
```

Comment:
```suggestion
    if bytes.len() >= size_of::<ArpHeader>() {
```
Header are only the first bytes of the `&[8]` there is the actual data after this